curated_transformers/tokenization/roberta_tokenizer.py

```diff
@@ -1,25 +1,59 @@
-from typing import Any, Type, TypeVar
+from typing import Any, Iterable, List, Type, TypeVar
 from cutlery import ByteBPEProcessor
 import json
 from pathlib import Path
 
 from .bbpe_tokenizer import ByteBPETokenizer
-from .tokenizer import PiecesWithIds, PostTokenizer
+from .tokenizer import PiecesWithIds, PostEncoder, PreDecoder
 
 
 # Only provided as typing.Self in Python 3.11+.
 Self = TypeVar("Self", bound="RobertaTokenizer")
 
 
-class RobertaPostTokenizer(PostTokenizer):
+class RobertaPreDecoder(PreDecoder):
     def __init__(
         self,
         *,
         processor: ByteBPEProcessor,
         bos_piece: str = "<s>",
         eos_piece: str = "</s>",
     ):
-        """Construct a RoBERTa post-tokenizer.
+        """Construct a RoBERTa pre-decoder.
+
+        processor (ByteBPEProcessor): The processor to wrap.
+        bos_piece (str): The piece to use to mark the beginning of a sequence.
+        eos_piece (str): The piece to use to mark the end of a sequence.
+        """
+        self.bos_piece = bos_piece
+        self.eos_piece = eos_piece
+        self.processor = processor
+
+    def __call__(self, input: Iterable[Iterable[int]]) -> List[List[int]]:
+        bos_id = self.processor.piece_id(self.bos_piece)
+        if bos_id is None:
+            raise ValueError(
+                f"RoBERTa piece encoder vocabulary doesn't contain '{self.bos_piece}' piece"
+            )
+
+        eos_id = self.processor.piece_id(self.eos_piece)
+        if eos_id is None:
+            raise ValueError(
+                f"RoBERTa piece encoder vocabulary doesn't contain '{self.eos_piece}' piece"
+            )
```

Comment:
Could we move these checks to the ctor? Or is it valid to pass an uninitialized/empty processor to it?

On a different note, we could move these checks into their own function and call them both in the pre-/post-decoder.

Reply:
Yeah, I think we need to allow to create blank tokenizers for `spacy-curated-transformers`. Though maybe we should make things nice here and hack around that later in `spacy-curated-transformers`? Even more with spaCy ❤️ PyTorch?

I'll factor these out to separate functions.

Reply:
I don't have a strong opinion as such, but having these checks happen inside the ctor in `curated-transformers` would indeed be nicer.

Reply:
Fixed.